### PR TITLE
Service user access

### LINKED ISSUE #332: Service user API access
For #329 and also to have easier management in Zappr Enterprise (e.g. automatically enable checks for all repos) we need to support authentication besides cookies.

Requirements:

* Authentication method is enabled and configured via configuration file
* Authentication methods offer clean API and can be replaced (maybe just passport plugins?)

As a first step we should support authentication via Zalando's Plan B and/or pre-shared secrets aka API keys.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #332 

* API endpoints respond also if you authenticate just as on Github: `Authorization: token XYZ` header.
* Token is validated against `/user` of Github API

### CODE REVIEW

**Comment by prayerslayer** [comment]
👍

**Comment by neremic** [comment]
:+1:

**Comment by prayerslayer** [comment]
👍

File: server/passport/passport-github-api.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    this.apiUrl = apiUrl
+    this.tokenType = tokenType
+    this.userAgent = userAgent
+    this.verify = verify
+  }
+
+  async authenticate(req) {
+    const {headers} = req
+    const AUTH_HEADER = headers['authorization']
+    if (!AUTH_HEADER) this.fail({status: 400, reason: 'Missing Authorization header'})
```
Comment (harti2006) [inline server/passport/passport-github-api.js]: wouldn't `401` be a more appropriate status?

Comment (prayerslayer) [inline server/passport/passport-github-api.js]: In this particular case, yes.

**Comment by harti2006** [comment]
:+1:

**Pull request merged**